### PR TITLE
Add error status handling

### LINKED ISSUE #7: Implement VideoState errored
Extend the VideoState interface with `errored` case and handle it internally

### PR DESCRIPTION
**Pull request by zzarcon**
closes #7 

Have a look @iloire

### CODE REVIEW

File: __tests__/index.tsx [hunk 1]
```diff
@@ -148,6 +148,14 @@ describe('VideoRenderer', () => {
     xit('should return the buffered value', () => {
       // TODO: fake multiple ranges
     });
+
+    it('should return error status when the video is errored', () => {
+      const {component, children} = setup();
+
+      component.find('video').simulate('error');
+
+      expect(component.state('status')).toEqual('errored');
```
Comment (zzarcon) [inline __tests__/index.tsx]: Just a /nit for the future, we `state` admits a property, instead of `state().status` :)

File: __tests__/index.tsx [hunk 2]
```diff
@@ -148,6 +148,14 @@ describe('VideoRenderer', () => {
     xit('should return the buffered value', () => {
       // TODO: fake multiple ranges
     });
+
+    it('should return error status when the video is errored', () => {
+      const {component, children} = setup();
+
+      component.find('video').simulate('error');
```
Comment (zzarcon) [inline __tests__/index.tsx]: @iloire so doing something like this with `play` didn't fired the onPlay handler right?

Comment (iloire) [reply __tests__/index.tsx]: This test is passing:

```
    it('should play the video when play action is called', () => {
      const {component} = setup({children});
      component.find('video').simulate('play');
      expect(component.state('status')).toEqual('playing');
    });
```

But this is not (testing the play action is what we want to do):

```
    it('should play the video when play action is called', () => {
      const {component} = setup({children});
      videoActions.play();
      expect(component.state('status')).toEqual('playing');
    });
```

Any ideas?

Comment (zzarcon) [reply __tests__/index.tsx]: Oh yeah, you are 💯right.
`videoActions.play()` is what we want to test for sure, and I don't think there is any easy way without manually firing that fake event from the element...

**Pull request merged**